app/api.py

```diff
@@ -53,6 +78,40 @@ def main_page(request: Request):
     )
 
 
+@app.post("/auth")
+async def authentication(form_data: Annotated[OAuth2PasswordRequestForm, Depends()], response: Response):
+    """
+    Authentication: provide username/password and get a bearer token in return
+
+    - **username**: Open Food Facts user_id (not email)
+    - **password**: user password (clear text, but HTTPS encrypted)
+
+    a **token** is returned
+    to be used in requests with usual "Authorization: bearer token" header
+    """
+    if "oauth2_server_url" not in settings.model_dump():
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="OAUTH2_SERVER_URL environment variable missing",
+        )
+
+    data = {"user_id": form_data.username, "password": form_data.password}
+    r = requests.post(settings.oauth2_server_url, data=data)  # type: ignore
+    if r.status_code == 200:
+        token = await create_token(form_data.username)
+        user: UserBase = {"user_id": form_data.username, "token": token}  # type: ignore
+        crud.create_user(db, user=user)  # type: ignore
+        return {"access_token": token, "token_type": "bearer"}
+    elif r.status_code == 403:
+        time.sleep(2)   # prevents brute-force
```

Comment:
you should use `asyncio.sleep` (https://docs.python.org/3/library/asyncio-task.html#asyncio.sleep) otherwise you're blocking your only thread

Reply:
ok thanks, done :heavy_check_mark: